storage.c

```diff
@@ -856,8 +856,30 @@ static void storage_compact_readback(void *storage, logger *l,
                         hdr->offset = io.offset;
                         rescues++;
                     } else {
-                        lost++;
-                        // TODO: re-alloc and replace header.
+                        // re-alloc and replace header.
+                        uint32_t flags;
+                        FLAGS_CONV(hdr_it, flags);
+                        item *new_it = do_item_alloc(ITEM_key(hdr_it), hdr_it->nkey, flags, hdr_it->exptime, sizeof(item_hdr));
+                        if (new_it == 0) {
+                            lost++;
+                        } else {
+                            // copy some item data.
+                            new_it->it_flags |= ITEM_HDR;
```

Comment:
pretty sure I need to copy over a few more flags:

token_sent/reserved/stale/key_binary and probably fetched.

probably just need to copy over it_flags wholesale from the old header but mask off or reset ITEM_LINKED.

Reply:
remove ITEM_LINKED and ITEM_SLABBED